src/rwe/Unit.cpp

```diff
@@ -95,19 +77,29 @@ namespace rwe
         }
 
         auto oldProgressEnergy = Energy((buildTimeCompleted * energyCost.value) / buildTime);
-        auto oldProgressMetal = Metal((buildTimeCompleted * metalCost.value) / buildTime);
+        auto oldProgressMetal  = Metal((buildTimeCompleted * metalCost.value) / buildTime);
 
         auto newBuildTimeCompleted = buildTimeCompleted + buildTimeContribution;
 
         auto newProgressEnergy = Energy((newBuildTimeCompleted * energyCost.value) / buildTime);
-        auto newProgressMetal = Metal((newBuildTimeCompleted * metalCost.value) / buildTime);
+        auto newProgressMetal  = Metal((newBuildTimeCompleted * metalCost.value) / buildTime);
 
         auto deltaEnergy = newProgressEnergy - oldProgressEnergy;
-        auto deltaMetal = newProgressMetal - oldProgressMetal;
+        auto deltaMetal  = newProgressMetal - oldProgressMetal;
 
         return BuildCostInfo{buildTimeContribution, deltaEnergy, deltaMetal};
     }
 
+    bool Unit::addRepairProgress(unsigned int repairTimeContribution)
+    {
+        if (buildTimeCompleted == buildTime)
```

Comment:
I hate this, but I was not able to find a cleaner solution fitting the current build logic.